Test failure in CI build 5069

The following test appears to have failed:

[#5069](https://circleci.com/gh/cockroachdb/cockroach/5069):

```
c106425462403429.14  127.0.0.1:0  18:00:25.732025  24.187µs    ·meta descriptor lookup       kv/dist_sender.go:614
c106425462403429.14  127.0.0.1:0  18:00:25.732057  4.086685ms  ·sending RPC                  kv/dist_sender.go:527
c106425462403429.14  127.0.0.1:0  18:00:25.732162  0           ··sending to 127.0.0.1:43562  rpc/send.go:171

==================
WARNING: DATA RACE
Write by main goroutine:
  flag.(*boolValue).Set()
      /usr/src/go/src/flag/flag.go:90 +0x7d
  github.com/cockroachdb/cockroach/cli.(*pflagValue).Set()
      <autogenerated>:1 +0x97
  github.com/spf13/pflag.(*FlagSet).setFlag()
      /go/src/github.com/spf13/pflag/flag.go:548 +0xb1
  github.com/spf13/pflag.(*FlagSet).parseLongArg()
      /go/src/github.com/spf13/pflag/flag.go:597 +0x4b5
  github.com/spf13/pflag.(*FlagSet).parseArgs()
--
  testing.(*M).Run()
      /usr/src/go/src/testing/testing.go:486 +0x136
  main.main()
      github.com/cockroachdb/cockroach/cli/_test/_testmain.go:68 +0x28c

Previous read by goroutine 61:
  github.com/cockroachdb/cockroach/util/log.(*loggingT).outputLogEntry()
      /go/src/github.com/cockroachdb/cockroach/util/log/clog.go:747 +0x28f
  github.com/cockroachdb/cockroach/util/log.AddStructured()
      /go/src/github.com/cockroachdb/cockroach/util/log/structured.go:39 +0x149
  github.com/cockroachdb/cockroach/util/log.logDepth()
      /go/src/github.com/cockroachdb/cockroach/util/log/log.go:65 +0x8f
  github.com/cockroachdb/cockroach/util/log.Infof()
      /go/src/github.com/cockroachdb/cockroach/util/log/log.go:85 +0x82
  github.com/cockroachdb/cockroach/server/status.(*NodeStatusMonitor).OnTrace()
      /go/src/github.com/cockroachdb/cockroach/server/status/monitor.go:225 +0xfa
--
I0728 18:00:36.348561 254 gossip/gossip.go:173  gossiping node descriptor node_id:2 address:<network:"tcp" address:"127.0.0.1:0" > attrs:<> 
I0728 18:00:36.352655 254 base/context.go:103  setting up TLS from certificates directory: test_certs
I0728 18:00:36.626841 254 gossip/server.go:113  gossip: received infostore with 3 info(s): info "node:1": node_id:1 address:<network:"tcp" address:"127.0.0.1:0" > attrs:<> , info "node:2": node_id:2 address:<network:"tcp" address:"127.0.0.1:0" > attrs:<> , info "local-key": local value
I0728 18:00:36.630853 254 gossip/client.go:155  gossip: received infostore with 1 info(s): info "remote-key": remote value
I0728 18:00:36.635384 254 gossip/server.go:113  gossip: received infostore with 1 info(s): info "remote-key": remote value
--- FAIL: TestClientGossip-4 (0.79s)
	<autogenerated>:32: gossip/client_test.go:109: condition failed to evaluate within 500ms: gossip/gossip.go:257: key "remote-key" does not exist or has expired
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
--
W0728 18:00:37.676949 254 gossip/gossip.go:527  missing sentinel gossip 0; assuming partition and reconnecting
W0728 18:00:37.692817 254 gossip/gossip.go:527  missing sentinel gossip 2; assuming partition and reconnecting
W0728 18:00:37.700600 254 gossip/gossip.go:527  missing sentinel gossip 0; assuming partition and reconnecting
W0728 18:00:37.707142 254 rpc/server.go:351  rpc: server cannot decode request: read tcp 127.0.0.1:36848: use of closed network connection
W0728 18:00:37.707982 254 rpc/server.go:408  rpc: write response failed
--- FAIL: TestClientDisconnectRedundant-4 (0.80s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestGossipInfoStore-4
--- FAIL: TestGossipInfoStore-4 (0.12s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestGossipGroupsInfoStore-4
--- FAIL: TestGossipGroupsInfoStore-4 (0.12s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestGossipGetNextBootstrapAddress-4
--- FAIL: TestGossipGetNextBootstrapAddress-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestMinGroupShouldInclude-4
--- FAIL: TestMinGroupShouldInclude-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestMaxGroupShouldInclude-4
--- FAIL: TestMaxGroupShouldInclude-4 (0.14s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestTypeMismatch-4
--- FAIL: TestTypeMismatch-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestSameKeyInserts-4
--- FAIL: TestSameKeyInserts-4 (0.13s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestGroupCompactAfterTTL-4
--- FAIL: TestGroupCompactAfterTTL-4 (0.13s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestGroups100Keys-4
--- FAIL: TestGroups100Keys-4 (0.31s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestSameKeySameTimestamp-4
--- FAIL: TestSameKeySameTimestamp-4 (0.12s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestSameKeyDifferentHops-4
--- FAIL: TestSameKeyDifferentHops-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestGroupGetInfo-4
--- FAIL: TestGroupGetInfo-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestGroupGetInfoTTL-4
--- FAIL: TestGroupGetInfoTTL-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestGroupWithStructVal-4
--- FAIL: TestGroupWithStructVal-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestPrefix-4
--- FAIL: TestPrefix-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestSort-4
--- FAIL: TestSort-4 (0.12s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestExpired-4
--- FAIL: TestExpired-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestIsFresh-4
--- FAIL: TestIsFresh-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestRegisterGroup-4
--- FAIL: TestRegisterGroup-4 (0.12s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestZeroDuration-4
--- FAIL: TestZeroDuration-4 (0.12s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestNewInfo-4
--- FAIL: TestNewInfo-4 (0.12s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestInfoStoreGetInfo-4
--- FAIL: TestInfoStoreGetInfo-4 (0.13s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestInfoStoreGetInfoTTL-4
--- FAIL: TestInfoStoreGetInfoTTL-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestAddInfoSameKeyLessThanEqualTimestamp-4
--- FAIL: TestAddInfoSameKeyLessThanEqualTimestamp-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestAddInfoSameKeyGreaterTimestamp-4
--- FAIL: TestAddInfoSameKeyGreaterTimestamp-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestAddInfoSameKeyDifferentHops-4
--- FAIL: TestAddInfoSameKeyDifferentHops-4 (0.12s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestAddGroupInfos-4
--- FAIL: TestAddGroupInfos-4 (0.10s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestCombine-4
--- FAIL: TestCombine-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestInfoStoreDelta-4
--- FAIL: TestInfoStoreDelta-4 (0.16s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestInfoStoreDistant-4
--- FAIL: TestInfoStoreDistant-4 (0.12s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestLeastUseful-4
--- FAIL: TestLeastUseful-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestCallbacks-4
--- FAIL: TestCallbacks-4 (0.12s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestRegisterCallback-4
--- FAIL: TestRegisterCallback-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestNodeSetMaxSize-4
--- FAIL: TestNodeSetMaxSize-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestNodeSetHasNode-4
--- FAIL: TestNodeSetHasNode-4 (0.12s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestNodeSetAddAndRemoveNode-4
--- FAIL: TestNodeSetAddAndRemoveNode-4 (0.12s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestNodeSetFilter-4
--- FAIL: TestNodeSetFilter-4 (0.12s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
=== RUN TestNodeSetAsSlice-4
--- FAIL: TestNodeSetAsSlice-4 (0.11s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
W0728 18:00:44.002134 254 rpc/server.go:408  rpc: write response failed
W0728 18:00:44.003892 254 rpc/server.go:408  rpc: write response failed
W0728 18:00:44.007545 254 rpc/server.go:408  rpc: write response failed
W0728 18:00:44.010637 254 rpc/server.go:408  rpc: write response failed
W0728 18:00:44.010958 254 rpc/server.go:408  rpc: write response failed
--- FAIL: TestConvergence-4 (1.65s)
	<autogenerated>:31: Test appears to have leaked an rpc client:
		github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2080e6a00, 0xc208090120, 0xc20800a460, 0x0, 0x0)
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:122 +0x1eee
		github.com/cockroachdb/cockroach/gossip.func·001()
			/go/src/github.com/cockroachdb/cockroach/gossip/client.go:77 +0x312
		github.com/cockroachdb/cockroach/util/stop.func·001()
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:79 +0x8c
		created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
			/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:80 +0x14c
		
--
			/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:34 +0x36
		github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc208038a00)
			/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x36
		main.main()
			github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x28d
FAIL
FAIL	github.com/cockroachdb/cockroach/gossip	7.933s
=== RUN TestParseResolverSpec-4
--- PASS: TestParseResolverSpec-4 (0.00s)
=== RUN TestGetAddress-4
--- PASS: TestGetAddress-4 (0.00s)
PASS
ok  	github.com/cockroachdb/cockroach/gossip/resolver	0.025s
?   	github.com/cockroachdb/cockroach/gossip/simulation	[no test files]
=== RUN TestKeySorting-4
--- PASS: TestKeySorting-4 (0.00s)
=== RUN TestMakeKey-4

```
Please assign, take a look and update the issue accordingly.

--- a/gossip/client_test.go
+++ b/gossip/client_test.go
@@ -26,7 +26,6 @@ import (
 	"github.com/cockroachdb/cockroach/util"
 	"github.com/cockroachdb/cockroach/util/hlc"
 	"github.com/cockroachdb/cockroach/util/leaktest"
-	"github.com/cockroachdb/cockroach/util/log"
 	"github.com/cockroachdb/cockroach/util/stop"
 )
 
@@ -84,15 +83,23 @@ func startGossip(t *testing.T) (local, remote *Gossip, stopper *stop.Stopper) {
 func TestClientGossip(t *testing.T) {
 	defer leaktest.AfterTest(t)
 	local, remote, stopper := startGossip(t)
+	disconnected := make(chan *client, 1)
+	client := newClient(remote.is.NodeAddr)
+
+	defer func() {
+		stopper.Stop()
+		if client != <-disconnected {
+			t.Errorf("expected client disconnect after remote close")
+		}
+	}()
+
 	if err := local.AddInfo("local-key", "local value", time.Second); err != nil {
 		t.Fatal(err)
 	}
 	if err := remote.AddInfo("remote-key", "remote value", time.Second); err != nil {
 		t.Fatal(err)
 	}
-	disconnected := make(chan *client, 1)
 
-	client := newClient(remote.is.NodeAddr)
 	// Use an insecure context. We're talking to unix socket which are not in the certs.
 	lclock := hlc.NewClock(hlc.UnixNano)
 	rpcContext := rpc.NewContext(nodeTestBaseContext, lclock, stopper)
@@ -107,12 +114,6 @@ func TestClientGossip(t *testing.T) {
 		}
 		return nil
 	})
-
-	stopper.Stop()
-	log.Info("done serving")
-	if client != <-disconnected {
-		t.Errorf("expected client disconnect after remote close")
-	}
 }
 
 // TestClientDisconnectRedundant verifies that the gossip server

